editor/src/components/canvas/controls/classname-select.tsx

```diff
@@ -37,18 +38,43 @@ import {
   getModifiableJSXAttributeAtPath,
   ModifiableAttribute,
 } from '../../../core/shared/jsx-attributes'
+import {
+  atomWithPubSub,
+  usePubSubAtomReadOnly,
+  usePubSubAtomWriteOnly,
+} from '../../../core/shared/atom-with-pub-sub'
 import { stripNulls } from '../../../core/shared/array-utils'
+import { mapToArray, mapValues } from '../../../core/shared/object-utils'
 
 interface TailWindOption {
   label: string
   value: string
+  attributes?: string[]
   categories?: string[]
 }
 
-const TailWindOptions: Array<TailWindOption> = AllTailwindClasses.map((className, index) => ({
-  label: className,
-  value: className,
-}))
+let TailWindOptions: Array<TailWindOption> = []
+let AttributeOptionLookup: { [attribute: string]: Array<TailWindOption> }
+
+async function loadTailwindOptions() {
+  TailWindOptions = mapToArray(
+    (attributes, className) => ({
+      label: className,
+      value: className,
+      attributes: attributes,
+    }),
+    ClassNameToAttributes,
+  )
+
+  AttributeOptionLookup = mapValues((classNames: Array<string>) => {
```

Comment:
This looks to be O(n * m * o) which could be pretty slow in the worst case.

Reply:
This only happens once on start up and was supposed to be wrapped in a Promise so that it didn't block first load. I'll fix that part but otherwise I think this is our best option rather than having to perform the lookup on each search

Reply:
Ok I've fixed this now. I also timed this and it is consistently taking ~65ms for me, which I'd say is acceptable since it's a one time thing and happens before the editor is responsive.